src/verify-executable-name-test.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "snap.h"
+#include "snap.c"
+
+#include <glib.h>
+
+static void test_verify_executable_name()
+{
+	// First, test the names we know are good
+	g_assert_true(verify_executable_name("snap.name.app"));
+	g_assert_true(verify_executable_name
+		      ("snap.network-manager.NetworkManager"));
+	g_assert_true(verify_executable_name("snap.f00.bar-baz1"));
+	g_assert_true(verify_executable_name("snap.foo.hook.bar"));
+	g_assert_true(verify_executable_name("snap.foo.hook.bar-baz"));
+
```

Comment:
Can you add a canary test for `"snap.name"`. At some point we may decide to abbreviate the security tag and this would be a nice test to catch that change.